src/main/java/bio/overture/ego/service/OAuthService.java

```diff
@@ -0,0 +1,40 @@
+package bio.overture.ego.service;
+
+import java.util.Map;
+import java.util.Optional;
+
+import com.github.scribejava.core.model.OAuth2AccessToken;
+import com.github.scribejava.core.model.OAuthRequest;
+import com.github.scribejava.core.model.Response;
+import com.github.scribejava.core.model.Verb;
+import com.github.scribejava.core.oauth.OAuth20Service;
+
+import org.springframework.boot.json.BasicJsonParser;
+import org.springframework.stereotype.Service;
+
+import bio.overture.ego.token.IDToken;
+
+@Service
+public class OAuthService {
+
+  public Optional<IDToken> getAuthInfo(String code, OAuth20Service service) {
+    // TODO: Fetch user info based on service type
+    try {
+      final OAuth2AccessToken accessToken = service.getAccessToken(code);
+
+      final OAuthRequest request = new OAuthRequest(Verb.GET,
+          "https://api.linkedin.com/v1/people/~:(email-address,first-name,last-name)?format=json");
+      service.signRequest(accessToken, request);
+      final Response response = service.execute(request);
+      BasicJsonParser parser = new BasicJsonParser();
+      Map<String, Object> result = parser.parseMap(response.getBody());
```

Comment:
As I mentioned in the previous line, this isn't a typed conversion. Having to deal with objects unsafely with lots of type checking is probably not the best approach. See my previous comment about using a class to represent the response.